contentcuration/contentcuration/management/commands/loadconstants.py

```diff
@@ -8,6 +8,112 @@
 from django.core.cache import cache
 logging = logmodule.getLogger(__name__)
 
+
+import os
+import le_utils
+
+
+BASE_URL = "https://raw.githubusercontent.com/learningequality/le-utils/master/le_utils/resources/{}"
```

Comment:
This line and lines 20-24 are now unused.

I have (had) some concerns about depending on internet connectivity for Studio operations (e.g. [this](https://github.com/learningequality/studio/pull/672/files#diff-1f72fcf306e83eef8d7438a69752b760R558)). Getting the resources from the local installation of `le-utils` is much better (the current approach in the code below). This way we have explicit versioning etc.

Reply:
Oops! I forgot to switch the function calls- switching them back. Regarding using le-utils from github or locally, one of the motivations behind referencing the github repo was that it allows us to get the latest constants without having to release a new version (allows for faster turnaround time for adding new languages, for instance). Using the local version, however, reduces the risk of there being a constant we don't want... What do you think would be the best approach?

Reply:
One concern I have here is that the imported le_utils, and the one the constants are pulled from, can be different. I think it would be better to somehow have loadconstants or git pull update the le_utils module as well, like adding le_utils as a git submodule. 

Maybe a quick short-term fix would be to try checking the version of online le_utils and warning (or erroring out) if it's newer than the local version, with a message asking people to update their le-utils package?

Reply:
My vote is stongly against the get-le-utils-from-github approach because it introduces an unnecessary "coupling" which we cannot control. It's not too complicated to release an le-utils and update the version pinned in studio (ideally with an `==` version). This way we can (1) release le-utils independently of studio, and (2) always make sure the same version is used by ricecooker, studio, and kolibri.

Reply:
It's not so much about effort as it is about ensuring quality of the library. When you have a library like le-utils that many clients use, you really should be testing any new release against all of the clients that use it, including Kolibri, ricecooker, and Studio. Forcing clients to rely on release versions to even finish landing a PR puts a lot of pressure to push out changes quickly, before doing that testing. We can go the route of pinning different versions to different clients, as I think happens now, but then you get into messy issues where updating the library version later on exposes bugs in the library that weren't caught earlier because, well, you only integration tested the library against one of its clients, not all of them. 

This, in fact, is actually a red flag that you have coupling problems - shared code gets too integrated with one particular client, and works poorly with other clients as a result. The way to avoid coupling problems is not to keep the code in its own separate space, but to ensure it's being tested as widely as possible with as many different clients as possible.

Generally, the most effective way of avoiding these issues is that the git sources for the client should always use the git sources for the library. (This is one common case for using git submodules.) All clients are then always then testing the latest code, giving us many more opportunities to catch bugs and incompatibilities in these libraries before a release is made.